pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -139,6 +140,13 @@ func (c *Controller) reconcile(qualifiedName util.QualifiedName) util.Reconcilia
 	// TODO(marun) Perform this defaulting in a webhook
 	corev1a1.SetFederatedTypeConfigDefaults(typeConfig)
```

Comment:
Since the `typeConfig` was set by `corev1a1.SetFederatedTypeConfigDefaults`, just curious which case will trigger the validation `typeconfig.CheckTypeConfigName` failed?

Reply:
If the target plural name and group are not set, then that function will ensure a valid type config.  The check is to ensure that if the target plural name and group are set, that the name matches.

Reply:
> If the target plural name and group are not set, 

When will this happen? Do you mean some mistake configuration of federated resource can cause this?

Reply:
There's nothing stopping someone from creating a type config with an arbitrary target that doesn't correspond to the name of the type config.  As per the linked issue, maintaining the correspondence between name and target configuration is essential to ensuring that a given federation control plane runs controllers for a given target type at most once.